### PR TITLE
Linting Tests

### PR DESCRIPTION
**Pull request by kej-jay**
Added tests so that some of the linting rules will be executed alongside the tests

### CODE REVIEW

**Comment by BenBals** [comment]
The code looks good, but we have to make all the tests pass to merge it, because we have an evergreen policy for development.

File: packages/AlgernonTests.package/ALGLinting.class/instance/testNoclassNameInSelector.st [hunk 1]
```diff
@@ -0,0 +1,3 @@
+code-style
+testNoclassNameInSelector
```
Comment (BenBals) [inline packages/AlgernonTests.package/ALGLinting.class/instance/testNoclassNameInSelector.st]: Please fix this typo

File: packages/AlgernonTests.package/ALGLinting.class/instance/testNoclassNameInSelector.st [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+	self ArbitraryCodeRule: #classNameInSelector
```
Comment (BenBals) [inline packages/AlgernonTests.package/ALGLinting.class/instance/testNoclassNameInSelector.st]: This errors: `Test not found`.

File: packages/AlgernonTests.package/ALGLinting.class/instance/testNoclassNameInSelector.st (hunk 1 above)
Comment (BenBals) [reply packages/AlgernonTests.package/ALGLinting.class/instance/testNoclassNameInSelector.st]: Please capitalize all tests correctly.

File: packages/AlgernonTests.package/ALGLinting.class/instance/testNodotAfterReturn.st [hunk 3]
```diff
@@ -0,0 +1,3 @@
+code-style
+testNodotAfterReturn
+	self ArbitraryCodeRule: #dotAfterReturn
```
Comment (BenBals) [inline packages/AlgernonTests.package/ALGLinting.class/instance/testNodotAfterReturn.st]: This errors: `Test not found`.

File: packages/AlgernonTests.package/ALGLinting.class/instance/testNocollectionProtocolRule.st [hunk 4]
```diff
@@ -0,0 +1,3 @@
+code-style
+testNocollectionProtocolRule
+	self ArbitraryCodeRule: #collectionProtocolRule
```
Comment (BenBals) [inline packages/AlgernonTests.package/ALGLinting.class/instance/testNocollectionProtocolRule.st]: This errors: `Test not found`.

File: packages/AlgernonTests.package/ALGLinting.class/instance/testNoinstVarInAllSubclasses.st [hunk 5]
```diff
@@ -0,0 +1,3 @@
+code-style
+testNoinstVarInAllSubclasses
+	self ArbitraryCodeRule: #instVarInAllSubclasses
```
Comment (BenBals) [inline packages/AlgernonTests.package/ALGLinting.class/instance/testNoinstVarInAllSubclasses.st]: This errors: `Test not found`.

File: packages/AlgernonTests.package/ALGLinting.class/instance/testNoinstanceVariableAccessIsConsistent.st [hunk 6]
```diff
@@ -0,0 +1,3 @@
+code-style
+testNoinstanceVariableAccessIsConsistent
+	self ArbitraryCodeRule: #instanceVariableAccessIsConsistent
```
Comment (BenBals) [inline packages/AlgernonTests.package/ALGLinting.class/instance/testNoinstanceVariableAccessIsConsistent.st]: This errors: `Test not found`.

File: packages/AlgernonTests.package/ALGLinting.class/instance/testNoinstanceVariableCapitalizationRule.st [hunk 7]
```diff
@@ -0,0 +1,3 @@
+code-style
+testNoinstanceVariableCapitalizationRule
+	self ArbitraryCodeRule: #instanceVariableCapitalizationRule
```
Comment (BenBals) [inline packages/AlgernonTests.package/ALGLinting.class/instance/testNoinstanceVariableCapitalizationRule.st]: This errors: `Test not found`.

File: packages/AlgernonTests.package/ALGLinting.class/instance/testNonoClassComments.st [hunk 8]
```diff
@@ -0,0 +1,3 @@
+code-style
+testNonoClassComments
```
Comment (BenBals) [inline packages/AlgernonTests.package/ALGLinting.class/instance/testNonoClassComments.st]: Please do not name things nono.

File: packages/AlgernonTests.package/ALGLinting.class/instance/testNonoClassComments.st [hunk 9]
```diff
@@ -0,0 +1,3 @@
...
+	self ArbitraryCodeRule: #noClassComments
```
Comment (BenBals) [inline packages/AlgernonTests.package/ALGLinting.class/instance/testNonoClassComments.st]: This errors: `Test not found`.

File: packages/AlgernonTests.package/ALGLinting.class/instance/testNotoDoRule.st [hunk 10]
```diff
@@ -0,0 +1,3 @@
+code-style
+testNotoDoRule
+	self ArbitraryCodeRule: #toDoRule
```
Comment (BenBals) [inline packages/AlgernonTests.package/ALGLinting.class/instance/testNotoDoRule.st]: This errors: `Test not found`.

**Review by BenBals — CHANGES_REQUESTED** [review]

**Comment by kej-jay** [comment]
> 
> 
> The code looks good, but we have to make all the tests pass to merge it, because we have an evergreen policy for development.

After digging a bit deeper I found out that there seems to be a problem with 9 of the tests because they could not be found. The linting rules regarding those tests all have an ID that does not start with '#smallLint' maybe there is an issue with the ArbitraryCodeRule?

**Comment by BenBals** [comment]
I don't think so. It was provided by the Lehrstuhl and makes to reference to #smallLint.

**Comment by kej-jay** [comment]
> 
> 
> I don't think so. It was provided by the Lehrstuhl and makes to reference to #smallLint.

Yes, but the 9 failing tests do not use smallLint

**Comment by BenBals** [comment]
You forgot a few colons. That's why to rules where not found.

**Pull request merged**